tensorrt_llm/_torch/auto_deploy/transform/library/kvcache.py

```
@@ -216,41 +269,98 @@ def _apply(
+                vswa_group_nodes[gi] = {}
+                for arg_name in std_arg_names:
+                    base = arg_name.removesuffix(host_suffix)
+                    if base in vswa_swappable_bases:
+                        is_host = arg_name.endswith(host_suffix)
+                        group_arg = f"{base}_g{gi}{host_suffix if is_host else ''}"
+                        vswa_group_nodes[gi][arg_name] = self._add_or_retrieve_input(
+                            gm, cm, group_arg
+                        )
+
+        # --- Pass 2: insert cached attention nodes with correct group metadata ---
+        for idx, (attn_node, qkv, cache_in_nodes, constants, group_idx) in enumerate(layer_infos):
```

Ruff (0.15.12)

[warning] Loop control variable `idx` not used within loop body

Rename unused `idx` to `_idx`

(B007)